lib/matplotlib/widgets.py

```diff
@@ -1973,20 +1991,45 @@ def __init__(self, ax, onselect, direction, minspan=None, useblit=False,
         self.direction = direction
 
         self.rect = None
-        self.pressv = None
+        self.visible = True
+        self._extents_on_press = None
 
         self.rectprops = rectprops
         self.onmove_callback = onmove_callback
         self.minspan = minspan
-        self.span_stays = span_stays
 
-        # Needed when dragging out of axes
-        self.prev = (0, 0)
+        self.maxdist = maxdist
+        # Deprecate `span_stays` in favour of interactive to be consistent
+        # with Rectangle, etc.
+        if span_stays:
+            interactive = True
```

Comment:
This should use `@_api.rename_parameter` to generate a standard deprecation message and to error out if e.g. the user passes both `span_stays` *and* `interactive`.

Reply:
Done